### PR TITLE
Implement get presignURL MU-182

### PR DESCRIPTION
**Pull request by capybbv**
https://somesandwich2.atlassian.net/browse/MU-182

### CODE REVIEW

File: src/src/main/java/com/hcmus/mentor/backend/infrastructure/fileupload/BlobStorage.java [hunk 1]
```diff
@@ -16,5 +24,6 @@ public interface BlobStorage {
 
   Boolean exists(String key);
 
-  String getUrl(String key);
+  String getUrl(String key)
+      throws ServerException, InsufficientDataException, ErrorResponseException, IOException, NoSuchAlgorithmException, InvalidKeyException, InvalidResponseException, XmlParserException, InternalException;
```
Comment (hieucckha) [inline src/src/main/java/com/hcmus/mentor/backend/infrastructure/fileupload/BlobStorage.java:28]: Separate this line into multiple lines

File: src/src/main/java/com/hcmus/mentor/backend/infrastructure/fileupload/impl/S3FileStorage.java [hunk 2]
```diff
@@ -31,40 +48,41 @@ public class S3FileStorage implements BlobStorage {
...
-    this.s3Presigner =
-        S3Presigner.builder()
-            .credentialsProvider(credentials)
-            .region(Region.of(s3Settings.RegionName))
-            .endpointOverride(URI.create(s3Settings.ServiceUrl))
-            .build();
+    this.minioClient = MinioClient.builder()
+        .endpoint(s3Settings.ServiceUrl)
+        .credentials(s3Settings.AccessKey, s3Settings.SecretKey)
+        .build();
```
Comment (hieucckha) [inline src/src/main/java/com/hcmus/mentor/backend/infrastructure/fileupload/impl/S3FileStorage.java:65]: Please move the minioClient below the `s3Client`

File: src/src/main/java/com/hcmus/mentor/backend/infrastructure/fileupload/impl/S3FileStorage.java [hunk 3]
```diff
@@ -31,40 +48,41 @@ public class S3FileStorage implements BlobStorage {
...
+    AwsCredentialsProvider credentials = StaticCredentialsProvider.create(
+        AwsBasicCredentials.create(s3Settings.AccessKey, s3Settings.SecretKey));
+
+    this.s3Client = S3Client.builder().credentialsProvider(credentials)
+        .region(Region.of(s3Settings.RegionName))
+        .endpointOverride(URI.create(s3Settings.ServiceUrl))
+        .forcePathStyle(s3Settings.ForcePathStyle).build();
+    this.s3Presigner = S3Presigner.builder().credentialsProvider(credentials)
+        .region(Region.of(s3Settings.RegionName))
+        .endpointOverride(URI.create(s3Settings.ServiceUrl)).build();
```
Comment (hieucckha) [inline src/src/main/java/com/hcmus/mentor/backend/infrastructure/fileupload/impl/S3FileStorage.java:76]: Are we use this presigner?

**Review by hieucckha — APPROVED** [review]

**Review by Dqvinh20 — APPROVED** [review]

File: src/src/main/java/com/hcmus/mentor/backend/infrastructure/fileupload/BlobStorage.java (hunk 1 above)
Comment (capybbv) [reply src/src/main/java/com/hcmus/mentor/backend/infrastructure/fileupload/BlobStorage.java:28]: Split this line into multiple lines but the code formatter has merged them. Should me separate and not reformat this file

**Pull request merged**